src/components/common/CreateUserWizard/StepUserName.tsx

```diff
@@ -37,63 +41,65 @@ const MSG = defineMessages({
 interface UsernameInputProps {
   username: string;
   disabled: boolean;
+  isValid: boolean;
 }
-const UsernameInput = ({ username, disabled }: UsernameInputProps) => (
+
+const UsernameInput = ({ username, disabled, isValid }: UsernameInputProps) => (
   <>
     <Input
       appearance={{ theme: 'fat' }}
       name="username"
       label={MSG.label}
-      formattingOptions={{ blocks: [100] }}
+      formattingOptions={{ blocks: [255] }}
       disabled={disabled}
     />
-    {username && isConfusing(username) && <ConfusableWarning />}
+    {username && isValid && isConfusing(username) && <ConfusableWarning />}
   </>
 );
 
-type StepValues = Pick<FormValues, 'username'>;
-type Props = Pick<
-  WizardStepProps<FormValues, StepValues>,
-  'nextStep' | 'wizardValues' | 'wizardForm'
->;
+type Props = WizardStepProps<FormValues, UserWizardStep2>;
 
-const StepUserName = ({ wizardValues, wizardForm }: Props) => {
+const StepUserName = ({
+  wizardValues,
+  wizardForm: { initialValues: defaultValues },
+}: Props) => {
   const navigate = useNavigate();
-
-  const transform = pipe(
-    mergePayload({
-      ...wizardValues,
-    }),
-    withMeta({ navigate }),
-  );
+  const transform = pipe(mergePayload(wizardValues), withMeta({ navigate }));
 
   return (
-    <ActionForm<StepValues>
+    <ActionHookForm<UserWizardStep2>
       submit={ActionTypes.USERNAME_CREATE}
       success={ActionTypes.USERNAME_CREATE_SUCCESS}
       error={ActionTypes.USERNAME_CREATE_ERROR}
       transform={transform}
       validationSchema={validationSchema}
-      {...wizardForm}
+      defaultValues={defaultValues}
     >
-      {({ dirty, isValid, isSubmitting, values }) => (
+      {({
+        getValues,
+        formState: { isValid, isSubmitting, isSubmitted, isDirty },
+      }) => (
         <UserStepTemplate
           heading={MSG.heading}
           description={MSG.description}
           descriptionValues={{ br: <br /> }}
           input={
-            <UsernameInput username={values.username} disabled={isSubmitting} />
+            <UsernameInput
+              username={getValues('username')}
+              disabled={isSubmitting || isSubmitted}
+              isValid={isValid}
+            />
           }
           button={
             <ContinueWizard
-              disabled={!isValid || !dirty || isSubmitting}
-              loading={isSubmitting}
+              disabled={!isValid || !isDirty || isSubmitting || isSubmitted}
+              loading={isSubmitting || isSubmitted}
```

Comment:
Using both `isSubmitting` and `isSubmitted` ensures the form shows its loading state while the saga executes in the background. `isSubmitting` is true while the form is being validated, and then `isSubmitted` is true once the saga has been initiated.